gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/NewDagManager.java

```diff
@@ -0,0 +1,393 @@
+    this.failedDagRetentionTime = timeUnit.toMillis(ConfigUtils.getLong(config, FAILED_DAG_RETENTION_TIME, DEFAULT_FAILED_DAG_RETENTION_TIME));
+    KillDagThread killDagThread = new KillDagThread(defaultJobStartSlaTimeMillis, this);
+    this.scheduledExecutorPool.scheduleAtFixedRate(killDagThread, 100L, 60L, TimeUnit.SECONDS);
+    this.failedDagStateStore = createDagStateStore(ConfigUtils.getConfigOrEmpty(config, FAILED_DAG_STATESTORE_PREFIX)
+        .withFallback(config), topologySpecMap);
+    setActive();
+  }
+
+  public synchronized void setActive() throws IOException {
+    this.dagStateStore = createDagStateStore(config, topologySpecMap);
+    this.failedDagIds = Collections.synchronizedSet(failedDagStateStore.getDagIds());
+    this.dagManagerMetrics.activate();
+    UserQuotaManager quotaManager = GobblinConstructorUtils.invokeConstructor(UserQuotaManager.class,
+        ConfigUtils.getString(config, ServiceConfigKeys.QUOTA_MANAGER_CLASS, ServiceConfigKeys.DEFAULT_QUOTA_MANAGER), config);
+    quotaManager.init(dagStateStore.getDags());
+    DagManager.FailedDagRetentionThread
+        failedDagRetentionThread = new DagManager.FailedDagRetentionThread(failedDagStateStore, failedDagIds, failedDagRetentionTime);
+    this.scheduledExecutorPool.scheduleAtFixedRate(failedDagRetentionThread, 0, retentionPollingInterval, TimeUnit.MINUTES);
+    loadDagFromDagStateStore();
+    ScheduledExecutorService houseKeepingThreadPool = Executors.newSingleThreadScheduledExecutor();
+    for (int delay = INITIAL_HOUSEKEEPING_THREAD_DELAY; delay < MAX_HOUSEKEEPING_THREAD_DELAY; delay *= 2) {
+      houseKeepingThreadPool.schedule(() -> {
+        try {
+          loadDagFromDagStateStore();
+        } catch (Exception e ) {
+          log.error("failed to sync dag state store due to ", e);
+        }}, delay, TimeUnit.MINUTES);
+    }
+  }
+
+  public DagStateStore createDagStateStore(Config config, Map<URI, TopologySpec> topologySpecMap) {
+    try {
+      Class<?> dagStateStoreClass = Class.forName(ConfigUtils.getString(config, DAG_STATESTORE_CLASS_KEY, FSDagStateStore.class.getName()));
+      return (DagStateStore) GobblinConstructorUtils.invokeLongestConstructor(dagStateStoreClass, config, topologySpecMap);
+    } catch (ReflectiveOperationException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public void addDag(String dagId, Dag<JobExecutionPlan> dag) {
+    this.dagManagementStateStore.addDag(dagId, dag);
+    this.dagManagementStateStore.addJobState(dagId, null);
+  }
+
+  @Override
+  public Dag<JobExecutionPlan> getDag(String dagId) {
+    return this.dagManagementStateStore.getDag(dagId);
+  }
+
+  @Override
+  public boolean containsDag(String dagId) {
+    return this.dagManagementStateStore.containsDag(dagId);
+  }
+
+  @Override
+  public List<Dag.DagNode<JobExecutionPlan>> getAllJobs() throws IOException {
+    return this.dagManagementStateStore.getAllJobs();
+  }
```

Comment:
who is supposed to invoke these, and why aren't they directly using `DagManagementStateStore`?